cli/cmd/test_connection.go

```diff
@@ -171,34 +185,146 @@ func testConnection(cmd *cobra.Command, args []string) error {
 	for i, client := range sourceClients {
 		pluginClient := plugin.NewPluginClient(client.Conn)
-		log.Info().Str("source", sources[i].VersionString()).Msg("Initializing source")
-		err := initPlugin(ctx, pluginClient, sources[i].Spec, false, invocationUUID.String())
+		log.Info().Str("source", sources[i].VersionString()).Msg("Testing source")
+		testResult, err := testPluginConnection(ctx, pluginClient, sources[i].Spec)
 		if err != nil {
-			initErrors = append(initErrors, fmt.Errorf("failed to init source %v: %w", sources[i].VersionString(), err))
-		} else {
-			log.Info().Str("source", sources[i].VersionString()).Msg("Initialized source")
+			allErrors = errors.Join(allErrors, fmt.Errorf("failed to test source %v: %w", sources[i].VersionString(), err))
+			continue
 		}
+		testResult.pluginRef = sources[i].VersionString()
+		testResult.pluginKind = "source"
+		testConnectionResults = append(testConnectionResults, *testResult)
 	}
 	for i, client := range destinationClients {
 		pluginClient := plugin.NewPluginClient(client.Conn)
-		log.Info().Str("destination", destinations[i].VersionString()).Msg("Initializing destination")
-		err := initPlugin(ctx, pluginClient, destinations[i].Spec, false, invocationUUID.String())
+		log.Info().Str("destination", destinations[i].VersionString()).Msg("Testing destination")
+		testResult, err := testPluginConnection(ctx, pluginClient, destinations[i].Spec)
 		if err != nil {
-			initErrors = append(initErrors, fmt.Errorf("failed to init destination %v: %w", destinations[i].VersionString(), err))
-		} else {
-			log.Info().Str("destination", destinations[i].VersionString()).Msg("Initialized destination")
+			allErrors = errors.Join(allErrors, fmt.Errorf("failed to test destination %v: %w", destinations[i].VersionString(), err))
+			continue
 		}
+		testResult.pluginRef = destinations[i].VersionString()
+		testResult.pluginKind = "destination"
+		testConnectionResults = append(testConnectionResults, *testResult)
 	}
 
-	allErrors := errors.Join(initErrors...)
 	status := cloudquery_api.SyncTestConnectionStatusCompleted
 	if allErrors != nil {
 		status = cloudquery_api.SyncTestConnectionStatusFailed
 	}
-	updateSyncTestConnectionStatus(ctx, log.Logger, status)
+	updateSyncTestConnectionStatus(context.Background(), log.Logger, status, testConnectionResults...)
+
+	log.Info().Any("testresults", testConnectionResults).Msg("Test connection completed")
+
+	maxLength := 0
+	for _, testResult := range testConnectionResults {
+		if len(testResult.pluginRef) > maxLength {
+			maxLength = len(testResult.pluginRef)
+		}
+	}
+
+	connFailures := make([]testConnectionResult, 0, len(testConnectionResults))
+	for i, testResult := range testConnectionResults {
+		if i == 0 {
+			cmd.Println("Test Connection Results")
+		}
+		cmd.Printf("%-12s %s %s", testResult.pluginKind, testResult.pluginRef, strings.Repeat(" ", maxLength-len(testResult.pluginRef)+1))
+		if testResult.Success {
+			cmd.Println("Success")
```

Comment:
```suggestion
			fmt.Println("Success")
```

Not saying it's better or worse, but this is how we do it in other places in the CLI

Reply:
<img width="942" alt="Screenshot 2024-05-17 at 15 28 57" src="https://github.com/cloudquery/cloudquery/assets/223029/4a1f68f0-61be-42f2-bb0b-6c64ecf4c56a">
I think this needs a PR :)